src/test/java/uk/ac/ebi/ampt2d/accessioning/commons/generators/monotonic/BlockManagerTest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.accessioning.commons.generators.monotonic;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.ac.ebi.ampt2d.accessioning.commons.generators.monotonic.exceptions.AccessionIsNotPending;
+import uk.ac.ebi.ampt2d.accessioning.commons.generators.exceptions.AccessionCouldNotBeGeneratedException;
+import uk.ac.ebi.ampt2d.accessioning.commons.generators.monotonic.persistence.entities.ContiguousIdBlock;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(SpringRunner.class)
+public class BlockManagerTest {
+
+    private static final String CATEGORY_ID = "category-id";
+    private static final String INSTANCE_ID = "instance-id";
+
+    @Test
+    public void noAvailableAccessionsIfNoBlocks() {
+        BlockManager manager = new BlockManager();
+        assertFalse(manager.hasAvailableAccessions(10));
+    }
+
+    @Test
+    public void availableAccessionWhenBlockHashBeenAdded() {
+        BlockManager manager = new BlockManager();
+        manager.addBlock(new ContiguousIdBlock(CATEGORY_ID, INSTANCE_ID, 0, 100));
+        assertTrue(manager.hasAvailableAccessions(10));
+        assertFalse(manager.hasAvailableAccessions(101));
```

Comment:
I don't completely understand why is this false. Is it because the new block is not contiguous?

Reply:
hasAvailableAccessions returns true or false if currently can generate X amount of accessions 10 isless than  100 that is the size of the only block on the manager, 101 is more than the block that it is loaded in the manager.